### PR TITLE
Resize options

### PR DESCRIPTION
**Pull request by matthewberryman**

### CODE REVIEW

File: lib/ImageProcessor.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ class ImageProcessor {
     resize(params) {
         return new Promise((resolve, reject) => {
             gm(this.image.getData())
+                .resize(params.width, params.height, params.resizeOption)
```
Comment (leonfancy) [inline lib/ImageProcessor.js:33]: Is it possible to make it optional?

File: lib/ImageProcessor.js [hunk 2]
```diff
@@ -30,6 +30,7 @@ class ImageProcessor {
...
                 .resize(params.width, params.height)
```
Comment (leonfancy) [inline lib/ImageProcessor.js]: You may need to delete this line.

File: readme.md [hunk 3]
```diff
@@ -77,6 +77,7 @@ Here is an example of configuration:
 - `ACL`: *private | public-read | public-read-write | authenticated-read | aws-exec-read | bucket-owner-read | bucket-owner-full-control*. Controls the permission of generated images.
 - `targetBucket`: (Optionally) Specify the bucket where you want to put the generated images. If omitted will use the same bucket as the source image.
 - `width/height`: It's better to just specify one of them, the other side can be resized based on the ratio.
+- `resizeOption`: For resizing images, this optional parameter can be set to one of the [ImageMagick geometry operators](https://www.imagemagick.org/script/command-line-processing.php#geometry).
```
Comment (leonfancy) [inline readme.md:83]: Could you give an example of this parameters?

**Review by leonfancy — COMMENTED** [review]
request change.

File: lib/ImageProcessor.js (hunk 1 above)
Comment (matthewberryman) [reply lib/ImageProcessor.js:33]: First of all, sorry, I had a stray duplicate resize call in from an incorrect merge resolution I did, which I fixed in https://github.com/urbanriskmap/image-lambda/commit/30421e3a3d60dbbbd16c07491ae1c7e82112b993.

This remaining single call to resize already works with params.resizeOption as optional. If it's not specified in the config, then here params.resizeOption will be null, and then the call to gm(image).resize(width, height, null) acts the same as resize(width, height) - something I tested just to be sure.

File: readme.md (hunk 3 above)
Comment (matthewberryman) [reply readme.md:83]: in https://github.com/slimfancy/image-lambda/pull/6/commits/0a1c1a474b08572517454b6e0300f25249e9e926 I've given an example in the config.json.example as well as bringing readme.md up-to-date with changes to that file.

**Pull request merged**